src/collection.js

```diff
@@ -521,8 +504,36 @@ export default class Collection {
    */
   importChanges(syncResultObject, changeObject) {
     return Promise.all(changeObject.changes.map(change => {
-      return this._importChange(change);
+      if (change.deleted) {
+        return Promise.resolve(change);
+      }
+      return this._decodeRecord("remote", change);
     }))
+      .then(decodedChanges => {
+        const changeIds = decodedChanges.map(change => change.id);
+        // XXX: this is not efficient at all.
+        return this.list({}, {includeDeleted: true})
```

Comment:
I had significant performance improvements by specifiying `{order: null}` as list params (with appropriate safety checks in `utils.js`)